lib/chargify_api_ares/resources/invoice.rb

```diff
@@ -9,27 +9,34 @@ class Payment < Base
     end
 
     def self.find_by_invoice_id(id)
-      find(:first, {:params => {:id => id}})
+      find(:first, { params:  { id: } })
     end
 
-    def self.find_by_subscription_id(id)
-      find(:all, {:params => {:subscription_id => id}})
+    def self.find_by_subscription_id(subscription_id)
+      find(:all, { params:  { subscription_id: } })
+    end
+
+    def self.find_by_subscription_and_date_period(subscription_id, start_date, end_date, extra_params = {})
+      find(
+        :all,
+        { params:  { subscription_id:, start_date:, end_date:, date_field: 'issue_date' }.merge(extra_params) },
+      )
```

Comment:
It will be used by hint-api in the following way:
```ruby
    invoices = Chargify::Invoice.find_by_subscription_and_date_period(
      hint_customer.vendor_subscription_id,
      start_date,
      end_date,
      { status: 'open,processing,paid' },
    )
```